### PR TITLE
Added resourceToString, resourceToByteArray, and resourceToURL methods to IOUtils

### PR DESCRIPTION
**Pull request by behrangsa**
Added `resourceToString(String name, Charset encoding)`, `resourceToByteArray(String name)`, and `resourceToURL(String name)` to read a classpath resource and convert it to String, byte array, and URL respectively.

### CODE REVIEW

File: src/test/java/org/apache/commons/io/IOUtilsTestCase.java [hunk 1]
```diff
@@ -1186,6 +1187,42 @@ private void testToString_URL(final String encoding) throws Exception {
...
+    @Test(expected = IOException.class) public void testResourceToByteArray_NonExistingResource() throws Exception {
+        IOUtils.resourceToByteArray("/non-existing-file.bin");
+    }
+
+    @Test(expected = Test.None.class) public void testResourceToURL_ExistingResourceAtRootPackage() throws Exception {
+        IOUtils.resourceToURL("/test-file-utf8.bin");
+    }
+
+    @Test(expected = Test.None.class) public void testResourceToURL_ExistingResourceAtSubPackage() throws Exception {
+        IOUtils.resourceToURL("/org/apache/commons/io/FileUtilsTestDataCR.dat");
```
Comment (ansell) [inline src/test/java/org/apache/commons/io/IOUtilsTestCase.java]: It is useful in positive test cases to verify things like the size or contents of the result to ensure that it isn't silently returning unuseful results. Ie, assertEquals("Wrong output size", expectedSize, result.size()) or similar as you can find above in this file.

Comment (behrangsa) [inline src/test/java/org/apache/commons/io/IOUtilsTestCase.java]: @ansell,

Done in c5af446ddc77b81801e451bd4e2c28cd685670ad.

**Comment by behrangsa** [comment]
Hi all,

Any comments, ideas, or suggestions? The JIRA issue is also created: https://issues.apache.org/jira/browse/IO-513#

Cheers,
Behrang

File: src/test/java/org/apache/commons/io/IOUtilsTestCase.java [hunk 2]
```diff
@@ -1186,6 +1187,89 @@ private void testToString_URL(final String encoding) throws Exception {
         }
     }
 
+    @Test public void testResourceToString_ExistingResourceAtRootPackage() throws Exception {
+        final long fileSize = new File(getClass().getResource("/test-file-utf8.bin").getFile()).length();
+        final String content = IOUtils.resourceToString("/test-file-utf8.bin", Charset.defaultCharset());
```
Comment (ansell) [inline src/test/java/org/apache/commons/io/IOUtilsTestCase.java]: Charset.defaultCharset here and in other tests seems wrong as it is locale/environment dependent. StandardCharsets.UTF_8 may be more appropriate.

Comment (behrangsa) [inline src/test/java/org/apache/commons/io/IOUtilsTestCase.java]: Agreed. I will fix it now.

**Comment by behrangsa** [comment]
Hi all,

I think this PR is good to go ;)

Cheers,
Behrang

**Comment by behrangsa** [comment]
This PR has been moved to: https://github.com/apache/commons-io/pull/28

**Pull request closed without merging**